### PR TITLE
KAFKA-3076: BrokerChangeListener should log the brokers in order

### PR DESCRIPTION
**Pull request by konradkalita**

### CODE REVIEW

**Comment by ijuma** [comment]
Thanks for the PR. Instead of sorting on every log statement, another approach is to sort it once and pass the sorted seq around. Have you considered this approach?

File: core/src/main/scala/kafka/controller/ReplicaStateMachine.scala [hunk 1]
```diff
@@ -352,24 +352,26 @@ class ReplicaStateMachine(controller: KafkaController) extends Logging {
...
+                .format(newBrokerIds.toSeq.sorted.mkString(","), deadBrokerIds.toSeq.sorted.mkString(","),
+                  controllerContext.liveBrokerIds.toSeq.sorted.mkString(",")))
               newBrokers.foreach(controllerContext.controllerChannelManager.addBroker(_))
               deadBrokerIds.foreach(controllerContext.controllerChannelManager.removeBroker(_))
-              if(newBrokerIds.size > 0)
+              if(newBrokerIds.nonEmpty)
                 controller.onBrokerStartup(newBrokerIds.toSeq)
-              if(deadBrokerIds.size > 0)
+              if(deadBrokerIds.nonEmpty)
                 controller.onBrokerFailure(deadBrokerIds.toSeq)
```
Comment (konradkalita) [inline core/src/main/scala/kafka/controller/ReplicaStateMachine.scala]: Sorting once seems cleaner I probably overcomplicated it, but it seems that we pass a Seq here only to convert it to a Set again, so maybe we just pass a Set?

Comment (ijuma) [inline core/src/main/scala/kafka/controller/ReplicaStateMachine.scala]: That is true, it's a bit unclear why we convert to a `Seq` and then immediately convert back to a `Set`. Another option is to introduce a method like `brokerSetToString` and have that method do the sorting.

**Comment by junrao** [comment]
Thanks for the patch. Yes, I agree with @ijuma . It seems it's simpler to just sort the broker ids once in BrokerChangeListener and pass them as Seq as they are now.

File: core/src/main/scala/kafka/controller/ReplicaStateMachine.scala [hunk 2]
```diff
@@ -365,14 +365,17 @@ class ReplicaStateMachine(controller: KafkaController) extends Logging {
               val brokerById = curBrokers.map(broker => broker.id -> broker).toMap
               val newBrokers = newBrokerIds.flatMap(brokerById.get)
               controllerContext.liveBrokers = curBrokers
+              val newBrokerIdsSorted = newBrokerIds.toSeq.sorted
+              val deadBrokerIdsSorted = deadBrokerIds.toSeq.sorted
+              val liveBrokerIdsSorted = controllerContext.liveBrokerIds.toSeq.sorted
```
Comment (ijuma) [inline core/src/main/scala/kafka/controller/ReplicaStateMachine.scala]: We should use `curBrokers` instead of `controllerContext.liveBrokerIds` here.

**Comment by ijuma** [comment]
LGTM

**Comment by junrao** [comment]
Thanks for the patch. LGTM

**Pull request closed without merging**